GitHubSearch/Services/RepositoryService.swift

```diff
@@ -27,16 +27,32 @@ final class RepositoryService {
 
     func fetchFavorites(completion: @escaping ([Repository]) -> Void) {
         let predicate = NSPredicate(format: "isFavorite = true")
-        realmService.fetch(Repository.self, predicate: predicate, completion: completion)
+        realmService.write { realm in
```

Comment:
We should not use `write()` directly. Instead place this logic in the `fetch()` in `RealmRepository` and here we will use just `realmService.fetch(Repository.self, predicate: predicate, completion: completion)` as it was before.